test/widgets/recent_dm_conversations_test.dart

```diff
@@ -49,10 +50,16 @@ Future<void> setupPage(WidgetTester tester, {
   await tester.pumpWidget(TestZulipApp(
     accountId: eg.selfAccount.id,
     navigatorObservers: navigatorObserver != null ? [navigatorObserver] : [],
-    child: const RecentDmConversationsPage()));
+    child: const HomePage()));
 
   // global store, per-account store, and page get loaded
   await tester.pumpAndSettle();
+
+  // Switch to direct messages tab.
+  await tester.tap(find.descendant(
+    of: find.byType(Center),
+    matching: find.byIcon(ZulipIcons.user)));
```

Comment:
Hmm, the use of `Center` is awkward — it's rather an implementation detail, both the fact that the desired copy of the icon does have a Center ancestor and the fact that the undesired copies don't.

I don't have a better idea to suggest, though, so I guess just leave it. No need for a TODO, even; really it just becomes actionable if the test breaks (because of some innocent change to either the nav bar, or how DMs appear in the inbox, or something else) and we have to find a different solution.

I feel like the API I'd want here is to be able to say "tap the such-and-such icon, specifically the one that's in the bottom part of the screen". But I don't see an easy way to do that with the existing `Finder` API.

Reply:
Yeah, I was a bit reluctant when writing this, but some alternatives would be customizing the initial view, adding a debug feature, or anything that only gets used in tests like this, which are not really realistic/useful. Leaving it seems fine to me.